frontend/app/entry.server.tsx

```diff
@@ -45,9 +45,10 @@ export default async function handleRequest(request: Request, responseStatusCode
   responseHeaders.set('Content-Security-Policy', contentSecurityPolicy);
   responseHeaders.set('Content-Type', 'text/html; charset=UTF-8');
   responseHeaders.set('Referrer-Policy', 'strict-origin-when-cross-origin');
-  responseHeaders.set('Set-Cookie', langCookie);
   responseHeaders.set('X-Content-Type-Options', 'nosniff');
   responseHeaders.set('X-Frame-Options', 'deny');
+  // .append() because there can be more than one cookie in a response
+  responseHeaders.append('Set-Cookie', langCookie);
```

Comment:
Have you tested the language switching with cookie since we may have more than one `Set-Cookie` for `langCookie`?

Reply:
Maybe the later `langCookie` set win. 🤷

Reply:
Remix doesn't clobber existing cookies when you return a `Set-Cookie` header in an action/loader: <https://github.com/remix-run/remix/blob/main/packages/remix-server-runtime/headers.ts#L35>